### ISSUE TITLE
year as a dataframe column instead of function argument?

### ISSUE THREAD

**Issue by erikarasnick** [geomarker-io/addNlcdData#2]

**Comment by cole-brokamp** [comment]
This is hard... I started to type up a response using `purrr::map2()` with a user-defined year column, but realized it wouldn't work because the functions rely on a data.frame input. I want to keep it easy and within the pipe workflow of adding data to existing data.frames, but this could make it difficult to use your own year column. Adding the year via a column would also create problems with making sure that column is named correctly, and I want to avoid "making up our own data structures" by assuming certain columns are available and named within the input data.frame.

Should we add an example of how to use a longitudinal address history using an NLCD year lookup table like I did here? https://github.com/geomarker-io/st_pm_hex/blob/2ef76fc5f29294e2ee0ccb732f2f7f06af927fb0/03_get_NLCD_data.R#L184-L209

We could expand the point example to have a randomly selected `year` to accompany each location and then use the year lookup table I linked above to get the right `nlcd_year` for each `year` (the closest year?). After getting all of the NLCD data needed for all possible `nlcd_year`s, then join in NLCD data to data.frame by `nlcd_year` and `nlcd_cell`
